lootbox/api.py

```diff
@@ -111,177 +135,314 @@ async def get_drop_handler(
+    limit: int = 10,
+    offset: int = 0,
+    db_session: Session = Depends(db.yield_db_session),
+) -> data.DropListResponse:
+    """
+    Get list of drops for a given dropper contract and claimant address.
+    """
+
+    if dropper_contract_address:
+        dropper_contract_address = Web3.toChecksumAddress(dropper_contract_address)
+
+    terminus_address = Web3.toChecksumAddress(terminus_address)
+
+    try:
+        results = actions.get_terminus_claims(
+            db_session=db_session,
+            dropper_contract_address=dropper_contract_address,
+            blockchain=blockchain,
+            terminus_address=terminus_address,
+            terminus_pool_id=terminus_pool_id,
+            active=active,
+            limit=limit,
+            offset=offset,
+        )
+    except NoResultFound:
+        raise DropperHTTPException(status_code=404, detail="No drops found.")
+    except Exception as e:
+        logger.error(
+            f"Can't get Terminus claims (blockchain={blockchain}, address={terminus_address}, pool_id={terminus_pool_id}): {e}"
+        )
+        raise DropperHTTPException(status_code=500, detail="Can't get claims")
+
+    return data.DropListResponse(drops=[result for result in results])
+
+
+@app.post("/drops/claims", response_model=data.DropCreatedResponse)
+async def create_drop(
+    register_request: data.DropRegisterRequest = Body(...),
+    db_session: Session = Depends(db.yield_db_session),
+) -> data.DropCreatedResponse:
+
+    """
+    Create a drop for a given dropper contract.
+    """
+
+    if register_request.terminus_address:
+        register_request.terminus_address = Web3.toChecksumAddress(
+            register_request.terminus_address
+        )
+
+    try:
+        claim = actions.create_claim(
+            db_session=db_session,
+            dropper_contract_id=register_request.dropper_contract_id,
+            title=register_request.title,
+            description=register_request.description,
+            claim_block_deadline=register_request.claim_block_deadline,
+            terminus_address=register_request.terminus_address,
+            terminus_pool_id=register_request.terminus_pool_id,
+            claim_id=register_request.claim_id,
+        )
+    except NoResultFound:
+        raise DropperHTTPException(status_code=404, detail="Dropper contract not found")
+    except Exception as e:
+        logger.error(f"Can't create claim: {e}")
+        raise DropperHTTPException(status_code=500, detail="Can't create claim")
+
+    return data.DropCreatedResponse(
+        dropper_claim_id=claim.id,
+        dropper_contract_id=claim.dropper_contract_id,
+        title=claim.title,
+        description=claim.description,
+        claim_block_deadline=claim.claim_block_deadline,
+        terminus_address=claim.terminus_address,
+        terminus_pool_id=claim.terminus_pool_id,
+        claim_id=claim.claim_id,
+    )
+
+
+@app.get("/drops/claimants", response_model=data.DropListResponse)
```

Comment:
@Andrei-Dolgolev : We should also add an endpoint like this:

```
curl -X POST /drops/claimants/diff -H "Content-Type: application/json" -d '[{"address": "<claimant_address>", "amount": <claimant_amount>}]'
```

This would calculate diff of request body list with the list of claimants in database.

It would display to the user the claimants who are in DB but not in list and the ones who are in list but not in DB.